core/src/toga/widgets/textinput.py

```diff
@@ -217,3 +222,22 @@ def validate(self):
         else:
             self._impl.set_error(error_message)
             return False
+
+    @property
+    def on_return(self):
+        """The handler to invoke when return is pressed.
+
+        Returns:
+            The function ``callable`` that is called when return is pressed.
+        """
+        return self._on_return
+
+    @on_return.setter
+    def on_return(self, handler):
+        """Set the handler to invoke when return is pressed.
+
+        Args:
+            handler (:obj:`callable`): The handler to invoke when return is pressed.
```

Comment:
We're moving away from Google-style docstrings; see the widgets that have had a completed audit (e.g., Button) for the preferred docstring style.